client/client.go

```diff
@@ -40,29 +40,21 @@ type DataExportClient interface {
 type Client struct {
 	HTTPClient *http.Client
 	Config     *config.Config
+	AuthFunc   func(*http.Request)
 }
 
 var _ DataExportClient = (*Client)(nil)
 
-// NewClient returns a Client initialized with http.DefaultClient and the
-// supplied apiToken.
-func NewClient(config *config.Config) *Client {
-	return &Client{
-		HTTPClient: http.DefaultClient,
-		Config:     config,
-	}
```

Comment:
I think this should stay here. Since you are trying to create a custom client, why don't you do something like:

```golang
myClient := client.NewClient(conf)
myClient.SetCustomAuthHeaders = func(req *http.Request) {
  // do custom auth
}
```

And then the client would have:

```golang
// no need to export
func (c *Client) setAuthHeaders(req *http.Request) {
  if c.SetCustomAuthHeaders != nil {
    c.SetCustomAuthHeaders(req)
    return
  }
  req.Header.Set("Authorization", "Basic "+c.Config.FsApiToken)
  for _, header := range c.Config.AdditionalHttpHeader {	
    req.Header.Set(header.Key, header.Value)	
  }
}.

Reply:
That may not cover the case where you want to actually change the client itself.  In client now, HTTPClient points to http.DefaultClient but in more complex auth schemes it does not.  For example, I use oauth2.Client which is also an *http.Client but has to be initialized differently.

If you are ok, with actually reaching into the Client object and mucking with the HTTPClient  field (currently it's uppercase) I am happy to do that and move the NewClient() back to client package.

Do you not think it's more "parallel" to the way we handle the Warehouse and Storage to have Client be a resource?

Reply:
No, since all the client has to do is match the `DataExportClient` interface. In truth, none of the fields in the `Client` struct should be exported.

It's actually better if you just implement your own client that implements the `DataExportClient` interface. You can pass that directly to the `NewHauser` function.

Reply:
See comment below about the new version.

If you would prefer (I asked in chat, but haven't heard back yet) I can duplicate the code from client.go and go the route of *just* being a different impl of `DataExportClient`.